code/modules/events/hive_threat.dm

```diff
@@ -0,0 +1,74 @@
+//A human groundside is identified as a threat to the hive, and the hive is rewarded for draining this target
+/datum/round_event_control/hive_threat
+	name = "Hive threat"
+	typepath = /datum/round_event/hive_threat
+	weight = 15
+	earliest_start = 10 MINUTES
+
+	gamemode_blacklist = list("Combat Patrol","Civil War","Sensor Capture")
+
+/datum/round_event_control/hive_threat/can_spawn_event(players_amt, gamemode)
+	for(var/mob/living/carbon/human/human in GLOB.alive_human_list)
+		if(HAS_TRAIT(human, TRAIT_HIVE_TARGET))
+			return FALSE //only one target at a time. Although maybe this isn't needed.
+	return ..()
+
+/datum/round_event/hive_threat
+	///The human target for this event
+	var/mob/living/carbon/human/hive_target
+	///The announcement sound
+	var/sound/queen_sound
```

Comment:
why is this a var on the actual event instead of in the one proc its used?

Reply:
I remember doing it for a reason. I uh, can't remember the reason. Originally I may have been making this as a more generic parent event I think.